addons/pos_restaurant/static/src/app/services/pos_store.js

```diff
@@ -325,33 +331,127 @@ patch(PosStore.prototype, {
+                                    orphanLine.preparationKey
+                                ]["quantity"];
+                        } else {
+                            destinationOrder.last_order_preparation_change.lines[
+                                adoptingLine.preparationKey
+                            ] = {
+                                ...order.last_order_preparation_change.lines[
+                                    orphanLine.preparationKey
+                                ],
+                                uuid: adoptingLine.uuid,
+                                transferredQty:
+                                    order.last_order_preparation_change.lines[
+                                        orphanLine.preparationKey
+                                    ]["quantity"],
+                            };
+                        }
+                    }
                     adoptingLine.merge(orphanLine);
                 } else {
+                    if (
+                        order.last_order_preparation_change.lines[orphanLine.preparationKey] &&
+                        !destinationOrder.last_order_preparation_change.lines[
+                            orphanLine.preparationKey
+                        ]
+                    ) {
+                        destinationOrder.last_order_preparation_change.lines[
+                            orphanLine.preparationKey
+                        ] = order.last_order_preparation_change.lines[orphanLine.preparationKey];
+                        orphanLine.setHasChange(true);
+                        orphanLine.toggleSkipChange();
+                    }
                     linesToUpdate.push(orphanLine);
                 }
             }
             linesToUpdate.forEach((orderline) => {
+                if (!orderline.origin_order_id) {
+                    orderline.origin_order_id = orderline.order_id.id;
+                }
                 orderline.order_id = destinationOrder;
             });
             this.setOrder(destinationOrder);
             if (destinationOrder?.id) {
                 this.addPendingOrder([destinationOrder.id]);
             }
-            await this.deleteOrders([order]);
+            if (!isMergedOrder) {
+                order.isTransferedOrder = true;
+                await this.deleteOrders([order]);
+            }
         }
         await this.setTable(destinationTable);
     },
+    async restoreOrdersToOriginalTable(order, destinationTableOrder) {
+        const orderlines = (destinationTableOrder.lines || []).reduce((acc, orderline) => {
+            if (orderline?.origin_order_id) {
+                const line = this.models["pos.order.line"].getBy("uuid", orderline.uuid);
+                acc.push(line);
+            }
+            return acc;
+        }, []);
+
+        for (const orderline of orderlines) {
+            if (orderline?.origin_order_id.id === order.id || order.table_id.child_ids.length) {
```

Comment:
`|| order.table_id.child_ids.length` I don't understand this condition. Can you explain?

Reply:
this condition is used to handled this case where 
more than 2 tables merged for example 3 tables with each 1 orderline 
table 1 --> table 2--> table 3.

now unmerge from `table 2` so now 2 orderlines should get back from `table 3`
but with this condition `orderline?.origin_order_id.id === order.id`
only 1 orderline will get back so `order.table_id.child_ids.length` this condition is used to determine 
that `middle table 2` have also child table with his so take that orderlines also.

see ref video: https://www.awesomescreenshot.com/video/34242062?key=fabf47ac594f977a10ae7dbb7a2b2933

Reply:
Also we will need this condition
```js
if (orderline?.origin_order_id.id === orderToExtract.id) {
    if (orderline.skip_change) {
        orderline.toggleSkipChange();
    }
    orderline.origin_order_id = null;
 }
 ```
 
 since we need to update that orderline only whose belongs to the current table (dragged table)